File: solutions/lua/flatten-array/1/flatten-array.lua
local function flatten(input)

    local res = {}
    for _, v in ipairs(input) do
        if type(v) ~= 'table' then
            if v then
                table.insert(res, v)
            end
        else
            for _, v1 in ipairs(flatten(v)) do
                table.insert(res, v1)
            end
        end
    end

    return res
end

return flatten

File: solutions/go/flatten-array/1/flatten_array.go
package flatten

func Flatten(nested interface{}) []interface{} {
	res := []interface{}{}
    switch l := nested.(type) {
        case []interface{}:
        	for _, e := range l {
                if e != nil {
                    inner := Flatten(e)
                    if len(inner) > 0 {
                        res = append(res, inner...)
                    }
                }
            }
        default:
        	res = append(res, l)
    }
    return res
}